.github/workflows/publish.yml

```diff
@@ -1,21 +1,30 @@
 name: Publish crates
 
 on:
+  # We only want to perform dry runs when this workflow runs on PRs.
+  pull_request:
+    branches: [master]
   workflow_dispatch:
     inputs:
       acvm-ref:
         description: The acvm reference to checkout
         required: true
+      publish:
+        description: Publish packages to crates.io
+        type: boolean
+        required: true
 
 jobs:
   publish:
     name: Publish in order
     runs-on: ubuntu-latest
+    env:
+      DRY_RUN_FLAG: ${{ !inputs.publish && "--dry-run" }}
```

Comment:
I hate to keep bearing bad news, but the run failed at https://github.com/noir-lang/acvm/actions/runs/5268520673/workflow

Reply:
Yep, no worries. This PR has been sitting at the back of my mind along with #325, I was going to address this this evening but got sidetracked with brillig foreign call things.

Reply:
Will try to get this working before we push out #355